### PR TITLE
fix: Use ember-fetch to avoid test failures

### PR DESCRIPTION
**Pull request by knownasilya**
closes #187

I'm not sure how to stub fetch, and fix the tests. Any guidance is appreciated.

Maybe something like http://www.wheresrhys.co.uk/fetch-mock/ ?

### CODE REVIEW

File: addon/authenticators/token.js [hunk 1]
```diff
@@ -180,24 +181,14 @@ export default Base.extend({
...
-        if (headers) {
-          Object.keys(headers).forEach(key => {
-            xhr.setRequestHeader(key, headers[key]);
-          });
-        }
-      }
-    });
+      body: JSON.stringify(data),
+      headers: modifiedHeaders
+    }).then((response) => response.json());
```
Comment (alechirsch) [inline addon/authenticators/token.js:192]: Wouldnt we want to return the response?

Comment (alechirsch) [reply addon/authenticators/token.js:192]: Just kidding, that does return it.

**Comment by knownasilya** [comment]
Also, I'd like to confirm that this PR does solve the original issue.

**Comment by jpadilla** [comment]
@knownasilya thanks for tackling this! `fetch-mock` looks like the simplest solution to me. Also, it seems like tests are failing in Travis due to some other thing unrelated to this PR, gah!

**Comment by tomclose** [comment]
I ran into this issue too. It makes it impossible to write robust acceptance tests that cover login and took a while to debug. It would be great to get this fix merged, so that others won't hit the same issue.

@knownasilya I've put up [a PR](https://github.com/jpadilla/ember-simple-auth-token/pull/207) that fixes the tests.

@jpadilla Is that all that's preventing this PR from being merged?

**Comment by TopsyCM** [comment]
I just set up a fork to implement this same fix. What needs doing to get this merged? I see the PR that @tomclose put up has a conflict, but is that all?

**Comment by jpadilla** [comment]
Hey y'all! Sorry for the delay here. With v3 recently out the door, would @knownasilya or anyone else want to pickup this up to move it forward?

**Comment by knownasilya** [comment]
I don't currently have time to work on the tests for this but can merge in any changes into my branch to continue the work.

**Comment by jpadilla** [comment]
Closed by #226. Thanks again ya'll!

**Pull request closed without merging**